### PR TITLE
ReactiveStreamsAdapters support for Single and Completable

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
ReactiveStreamsAdapters currently only supports Publisher type. Users
can manually convert from Single and Completable to Publisher, but this
isn't a streamlined approach.

Modifications:
- Add conversion methods for Single and Completable to RS Publisher in
ReactiveStreamsAdapters

Result:
Direct conversions from all concurrent-api types to RS Publisher.

### CODE REVIEW

File: servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java [hunk 1]
```diff
@@ -73,6 +80,54 @@ private ReactiveStreamsAdapters() {
...
+     */
+    public static <T> org.reactivestreams.Publisher<T> toReactiveStreamsPublisher(Single<T> source) {
+        return toReactiveStreamsPublisher(source.toPublisher());
+    }
+
+    /**
+     * Converts the passed {@link SingleSource} to a
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm">Reactive Streams</a>
+     * {@link org.reactivestreams.Publisher}.
+     * @param source {@link Single} to convert to a {@link org.reactivestreams.Publisher}.
```
Comment (idelpivnitskiy) [inline servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java:99]: `Single` -> `SingleSource`

File: servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java [hunk 2]
```diff
@@ -73,6 +80,54 @@ private ReactiveStreamsAdapters() {
...
+     */
+    public static <T> org.reactivestreams.Publisher<T> toReactiveStreamsPublisher(SingleSource<T> source) {
+        return toReactiveStreamsPublisher(fromSource(source).toPublisher());
+    }
+
+    /**
+     * Converts the passed {@link Completable} to a
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm">Reactive Streams</a>
+     * {@link org.reactivestreams.Publisher}.
+     * @param source {@link Single} to convert to a {@link org.reactivestreams.Publisher}.
```
Comment (idelpivnitskiy) [inline servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java:111]: `Single` -> `Completable`

File: servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java [hunk 3]
```diff
@@ -73,6 +80,54 @@ private ReactiveStreamsAdapters() {
...
+     */
+    public static <T> org.reactivestreams.Publisher<T> toReactiveStreamsPublisher(Completable source) {
+        return toReactiveStreamsPublisher(source.toPublisher());
+    }
+
+    /**
+     * Converts the passed {@link CompletableSource} to a
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm">Reactive Streams</a>
+     * {@link org.reactivestreams.Publisher}.
+     * @param source {@link Single} to convert to a {@link org.reactivestreams.Publisher}.
```
Comment (idelpivnitskiy) [inline servicetalk-concurrent-reactivestreams/src/main/java/io/servicetalk/concurrent/reactivestreams/ReactiveStreamsAdapters.java:123]: `Single` -> `CompletableSource`

**Review by idelpivnitskiy — APPROVED** [review]
What about `from*` methods for `Single` and `Completable`? I think we need `Publisher.completeOrError()` operator, similar to `Publisher.firstOrError()`.

**Comment by Scottmitch** [comment]
Since there is no type specialization for RS `Publisher` to indicate single/completable semantics I think the "from" side makes more sense to leverage conversion operators after the fact bcz there are different ways todo this (e.g. firstOrElse, firstOrError, etc.).

**Comment by Scottmitch** [comment]
Followup PR for `Publisher#completableOrError` https://github.com/apple/servicetalk/pull/1256

**Pull request merged**